Tools/llm-tool/LLMTool.swift

```diff
@@ -29,19 +40,67 @@ struct LLMArguments: ParsableArguments {
     @Option(name: .shortAndLong, help: "The sampling temperature")
     var temperature: Float = 0.6
 
-    @Option(name: .shortAndLong, help: "The top p sampling")
+    @Option(name: .long, help: "The top p sampling")
     var topP: Float = 0.9
 
-    @Option(name: .shortAndLong, help: "The penalty factor for repeating tokens")
+    @Option(name: .long, help: "The penalty factor for repeating tokens")
     var repetitionPenalty: Float = 1.0
 
-    @Option(name: .shortAndLong, help: "The number of tokens to consider for repetition penalty")
+    @Option(name: .long, help: "The number of tokens to consider for repetition penalty")
     var repetitionContextSize: Int = 20
 
     @Option(name: .long, help: "The PRNG seed")
     var seed: UInt64 = 0
 
-    @Flag(help: "Show memory stats")
+    var generateParameters: GenerateParameters {
+        GenerateParameters(
+            temperature: temperature, topP: topP, repetitionPenalty: repetitionPenalty,
+            repetitionContextSize: repetitionContextSize)
+    }
+
+    func tokenizePrompt(configuration: ModelConfiguration, tokenizer: Tokenizer) -> (String, [Int])
+    {
+        MLXRandom.seed(seed)
+
+        let prompt = configuration.prepare(prompt: self.prompt)
+        let promptTokens = tokenizer.encode(text: prompt)
+
+        return (prompt, promptTokens)
+    }
+
+    func generate(promptTokens: [Int], model: LLMModel, tokenizer: Tokenizer) async
+        -> GenerateResult
+    {
+        // track how much we have printed
+        var printed = 0
+
+        return await LLM.generate(
+            promptTokens: promptTokens, parameters: generateParameters,
+            model: model, tokenizer: tokenizer
+        ) { tokens in
```

Comment:
Wrap the generic generation in something useful for command line tools.  Of course callers can use `LLM.generate()` directly if they need something custom.